g4me/analysis/PrepareTracksForMatchingAndFit_embedding.C

```diff
@@ -0,0 +1,330 @@
+      trackIdHitLayer1 = io_underlaying.hits.trkid[idHitLayer1];
+
+      for (int iHitLayer2=0; iHitLayer2<nHits_MIDLayer2; iHitLayer2++) {
+
+	       idHitLayer2 = arrayHitID_MIDLayer2[iHitLayer2];
+         posHitMID2.SetXYZ(gRandom->Gaus(io_underlaying.hits.x[idHitLayer2],resolutionMID),gRandom->Gaus(io_underlaying.hits.y[idHitLayer2],resolutionMID),gRandom->Gaus(io_underlaying.hits.z[idHitLayer2],resolutionMID));
+	       trackIdHitLayer2 = io_underlaying.hits.trkid[idHitLayer2];
+
+	       if (trackletSel->IsMIDTrackletSelected(posHitMID1,posHitMID2,kFALSE)) {
+
+	          if (trackIdHitLayer1==trackIdHitLayer2) trackletID = trackIdHitLayer1;
+	          else                                    trackletID = -1;
+
+	          TClonesArray trackletMIDpos("TVector3");
+	          TClonesArray trackletMIDcov("TMatrixDSym");
+
+	          new (trackletMIDpos[trackletMIDpos.GetEntries()]) TVector3(posHitMID1);
+	          new (trackletMIDpos[trackletMIDpos.GetEntries()]) TVector3(posHitMID2);
+	          new (trackletMIDcov[trackletMIDcov.GetEntries()]) TMatrixDSym(covMID);
+	          new (trackletMIDcov[trackletMIDcov.GetEntries()]) TMatrixDSym(covMID);
+
+	          new (trackCandidatesHitPosMID[nPreparedTrackletsMID]) TClonesArray(trackletMIDpos);
+	          new (trackCandidatesHitCovMID[nPreparedTrackletsMID]) TClonesArray(trackletMIDcov);
+
+	          idTrackMID.emplace_back(trackletID);
+
+	          nPreparedTrackletsMID++;
+
+	      }
+      }
+    }
+
+    // end of loop over pythia tree
+    //--------------------------------------------------------------------------
+    // Loop over muon hits
+    for (int iHit=0; iHit<io_signal.hits.n; iHit++) {
+
+      // filling arrays of hit IDs from MID layers (coming from any charged tracks)
+
+      auto trackID = io_signal.hits.trkid[iHit];
+
+      if (!(IsTrackCharged(&io_signal, trackID))) continue;
+
+      mom.SetXYZ(io_signal.hits.px[iHit],io_signal.hits.py[iHit],io_signal.hits.pz[iHit]);
+      if (mom.Mag() < hitMinP) continue;
+
+      if (io_signal.hits.lyrid[iHit] == idLayerMID1) arrayHitID_MIDLayer1[nHits_MIDLayer1++] = iHit;
+      if (io_signal.hits.lyrid[iHit] == idLayerMID2) arrayHitID_MIDLayer2[nHits_MIDLayer2++] = iHit;
+
+      // filling arrays of hits from ITS tracks (only for interesting tracks: charged and primary).
+      // Hits from ITS are by definition all the hits having radius < rMaxITS
+      if (!(IsTrackInteresting(&io_signal, trackID))) continue;
+
+      pos.SetXYZ(gRandom->Gaus(io_signal.hits.x[iHit],resolutionITS),gRandom->Gaus(io_signal.hits.y[iHit],resolutionITS),gRandom->Gaus(io_signal.hits.z[iHit],resolutionITS));
+      if (pos.Perp() < rMaxITS) {
+	       new ((allTracksHitPosITS.at(trackID))[(allTracksHitPosITS.at(trackID)).GetEntries()]) TVector3(pos);
```

Comment:
When used to retrieve a specific element of allTracksHitPosITS and allTracksHitCovITS, trackID should be shifted by io_underlying.tracks.n, right?